### PR TITLE
[tests-only] added spaces examples for TUS related upload features

### LINKED ISSUE #3028: use space dav path for TUS upload tests
### Description 
use spaces DAV path for TUS upload acceptance tests

**tests/acceptance/features/apiWebdavUploadTUS**
├── checksums.feature
├── creationWithUplaodExtension.feature
├── lowLevelCreationExtension.feature
├── lowLevelUpload.feature
├── optionsRequest.feature
├── uploadFile.feature
├── uploadFileMtime.feature
├── uploadFileMtimeShares.feature
├── uploadToMoveFolder.feature
├── uploadToNonExistingFolder.feature
└── uploadToShare.feature

### Related
- https://github.com/owncloud/ocis/issues/2854

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- use spaces DAV variant for TUS related upload features

## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/3028
- Reported https://github.com/owncloud/ocis/issues/3048
- Reported https://github.com/owncloud/ocis/issues/3050

## How Has This Been Tested?

- :robot: 
- https://github.com/owncloud/ocis/pull/3044

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiWebdavUploadTUS/checksums.feature [hunk 1]
```diff
@@ -34,6 +40,10 @@ Feature: checksums
       | dav_version |
       | old         |
       | new         |
+    @personalSpace
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavUploadTUS/checksums.feature:43]:
```suggestion

    @personalSpace
```

Comment (kiranparajuli589) [reply tests/acceptance/features/apiWebdavUploadTUS/checksums.feature:43]: fixed

File: tests/acceptance/features/bootstrap/ChecksumContext.php [hunk 2]
```diff
@@ -223,48 +225,56 @@ public function theWebdavChecksumShouldMatch(string $expectedChecksum):void {
 			__METHOD__ . " parsed XML parsed0 does not have key value"
 		);
 		$parsed0Value = $parsed0['value'];
+		Assert::assertIsArray($parsed0Value, __METHOD__ . " parsed XML parsed0Value is not an array");
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/ChecksumContext.php:228]: IMO those error messages will not help when seeing them in a test-run output.
```suggestion
		Assert::assertIsArray($parsed0Value, __METHOD__ . " could not find checksum in parsed XML expecting `$parsed[0]['value']` to be an array but received '" . $parsed0Value . "'");
```

similar in the following code

File: tests/TestHelpers/WebDavHelper.php [hunk 3]
```diff
@@ -671,30 +673,44 @@ public static function getMtimeOfFileinPublicLinkShare(
...
+			"0",
+			"files",
+			$davPathVersionToUse
 		);
 		$responseXmlObject = HttpRequestHelper::getResponseXml(
 			$response,
 			__METHOD__
 		);
 		$xmlpart = $responseXmlObject->xpath("//d:getlastmodified");
+		Assert::assertIsArray(
```
Comment (individual-it) [inline tests/TestHelpers/WebDavHelper.php:705]: do we need this assertion if we also have `Assert::assertArrayHasKey(`?

File: tests/TestHelpers/WebDavHelper.php [hunk 4]
```diff
@@ -671,30 +673,44 @@ public static function getMtimeOfFileinPublicLinkShare(
...
+			$xmlpart,
+			__METHOD__ . " last modified expression is not found in the XML"
+		);
+		Assert::assertArrayHasKey(
+			0,
+			$xmlpart,
+			__METHOD__ . " XML part does not have key 0"
```
Comment (individual-it) [inline tests/TestHelpers/WebDavHelper.php:712]: please be more descriptive in the error message

**Review by individual-it — CHANGES_REQUESTED** [review]
I'm missing the `@skipOnOc10` tags. Do we still need them?

File: tests/TestHelpers/WebDavHelper.php (hunk 3 above)
Comment (kiranparajuli589) [reply tests/TestHelpers/WebDavHelper.php:705]: sometimes, input for `Assert::assertArrayHasKey()` is not an array, so it throws strange errors.

**Comment by kiranparajuli589** [comment]
> I'm missing the `@skipOnOc10` tags. Do we still need them?

all of the features in `apiWebdavUploadTUS` suites already have `@skipOnOcV10` tag at top

File: tests/TestHelpers/WebDavHelper.php (hunk 4 above)
Comment (kiranparajuli589) [reply tests/TestHelpers/WebDavHelper.php]: fixed

File: tests/acceptance/features/bootstrap/ChecksumContext.php (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/ChecksumContext.php]: fixed

File: tests/TestHelpers/WebDavHelper.php [hunk 5]
```diff
@@ -671,30 +673,44 @@ public static function getMtimeOfFileinPublicLinkShare(
...
 		);
 		$xmlpart = $responseXmlObject->xpath("//d:getlastmodified");
+		Assert::assertIsArray(
+			$xmlpart,
+			__METHOD__ . " last modified expression is not found in the XML. Expected 'xmlPart' to be an array but, found: " . $xmlpart
+		);
+		Assert::assertArrayHasKey(
+			0,
+			$xmlpart,
+			__METHOD__ . " XML part does not have key 0. Expected a value at index 0 of 'xmlPart' but, found: " . $xmlpart[0]
```
Comment (individual-it) [inline tests/TestHelpers/WebDavHelper.php:712]:
```suggestion
			__METHOD__ . " XML part does not have key 0. Expected a value at index 0 of 'xmlPart' but, found: " . $xmlpart
```
the key does not exist if that assertion fires

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioning-v1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/54/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiMain-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/53/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFavorites-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/52/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareReshareToRoot3-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/77/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareReshareToRoot2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/75/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavUpload1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/98/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot1-10.8.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/104/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareManagementToRoot-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/63/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareOperationsToShares2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/70/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavMove2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/93/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareUpdateToRoot-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/79/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot2-latest-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/106/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot1-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/102/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot2-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/105/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot2-10.8.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34463/107/1

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34464/172/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34466/172/1

**Review by individual-it — APPROVED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharePublicLink1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34476/71/1

File: tests/TestHelpers/WebDavHelper.php [hunk 6]
```diff
@@ -671,30 +673,44 @@ public static function getMtimeOfFileinPublicLinkShare(
...
+			$davPathVersionToUse
 		);
 		$responseXmlObject = HttpRequestHelper::getResponseXml(
 			$response,
 			__METHOD__
 		);
 		$xmlpart = $responseXmlObject->xpath("//d:getlastmodified");
+		Assert::assertIsArray(
+			$xmlpart,
+			__METHOD__ . " last modified expression is not found in the XML. Expected 'xmlPart' to be an array but, found: " . $xmlpart
```
Comment (individual-it) [inline tests/TestHelpers/WebDavHelper.php:707]: Tests fail now with `Notice: Array to string conversion in /drone/src/tests/TestHelpers/WebDavHelper.php line 707`
for e.g test tests/acceptance/features/apiSharePublicLink1/createPublicLinkShareOc10Issue37605.feature:9
see https://drone.owncloud.com/owncloud/core/34476/71/13

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliBackground-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34478/114/1

**Review by phil-davis — APPROVED** [review]
LGTM. There have been other PRs merged to master this morning, so you might like to rebase this and let CI run, just to be sure that everything is OK. Then merge.

**Comment by phil-davis** [comment]
@kiranparajuli589 you can mark "Ready for Review" and merge when you are ready. Then make the next oCIS commit id bump.

**Comment by phil-davis** [comment]
"This branch is 1 commit ahead of master. " - looks good.

**Comment by kiranparajuli589** [comment]
test PR passed https://drone.owncloud.com/owncloud/ocis/9059

**Pull request merged**